src/html/html-document.ts

```diff
@@ -63,17 +65,35 @@ export class ParsedHtmlDocument extends ParsedDocument<ASTNode, HtmlVisitor> {
         end: {line: location.endTag.line - 1, column: location.endTag.col - 1}
       };
     }
-    // text node value can provide accurate line and column count
+
+    // text node
     if (parse5.treeAdapters.default.isTextNode(node)) {
       const lines = (node.value || '').split(/\n/);
-      const lastLine = lines[lines.length - 1];
-      const endColumn = lines.length === 1 ? location.col + lastLine.length - 1 : lastLine.length;
+      const lastLineLength = lines[lines.length - 1].length;
+      const endColumn = lines.length === 1 ? location.col + lastLineLength - 1
+          : lastLineLength;
+      return {
+        file: this.url,
+        start: {line: location.line - 1, column: location.col - 1},
+        end: {line: location.line + lines.length - 2, column: endColumn}
+      };
+    }
+
+    // comment node
+    if (parse5.treeAdapters.default.isCommentNode(node)) {
+      const lines = (parse5.treeAdapters.default.getCommentNodeContent(node) ||
+          '').split(/\n/);
+      const lastLineLength = lines[lines.length - 1].length;
+      const endColumn = lines.length === 1 ? location.col + lastLineLength + 6
```

Comment:
The +6 vs +3 is described in the comment for new _sourceRangeForCommentNode header here https://github.com/Polymer/polymer-analyzer/pull/344/files#diff-d9049ec942cc8c27254037cf14faa624R64